File: csharp/Auction-TestDesign-Kata/Test-Auction/AuctionEventListenerApprovalSpy.cs
using Auction;

namespace Test_Auction_TestDesign_Kata;

public class AuctionEventListenerApprovalSpy : IAuctionEventListener
{
    private string log = "";

    public void AuctionClosed()
    {
        log += "auctionClosed";
    }

    public void CurrentPrice(int price)
    {
        log += "currentPrice " + price;
    }

    public void AuctionFailed()
    {
        log += "auctionFailed";
    }

    public override string ToString()
    {
        return log;
    }
}

File: kotlin/src/test/kotlin/org/sammancoaching/AuctionEventListenerApprovalSpy.kt
package org.sammancoaching

internal class AuctionEventListenerApprovalSpy(val log: StringBuilder) : AuctionEventListener {
    override fun auctionClosed() {
        log.append("auctionClosed")
    }

    override fun currentPrice(price: Int) {
        log.append("currentPrice " + price)
    }

    override fun auctionFailed() {
        log.append("auctionFailed")
    }

    override fun toString(): String {
        return log.toString()
    }
}